libs/frontend/feature-preview/src/lib/preview-component.ts

```diff
@@ -64,21 +66,45 @@ export const PreviewComponent = () => {
             return h('a', renderProps, children)
           } else {
             // Relative path, forward query string and hash.
+
             // In vue-router, users can pass in an optional `href` prop to <router-link> to override
-            // the href generated by the required prop `to`.
-            // If `to` and `href` are both present, <router-link> will use the value of `href` prop as
-            // the href of the rendered <a>, but upon click, users will actually be navigated to the
-            // href generate by `to`.
-            // To avoid confusion, it's better to use `to` alone here.
+            // the href generated by the required prop `to`. If `to` and `href` are both present,
+            // `href` will be used to render <a>, but upon click, users will actually be navigated to
+            // the href generate by `to`. To avoid confusion, it's better to use `to` alone here.
+
+            // Since users may provide href with custom hash and query, we have to extract those values
+            // from the `href` prop and merge them with editor hash and query. User-provided hash&query
+            // will take precedence over editor hash&query.
+
+            const linkURL = hrefToUrl(hrefProp, '/preview')
+            const {
+              pathname: previewPathName,
+              search: userSearch,
+              hash: userHash,
+            } = linkURL
+            const { search: editorSearch, hash: editorHash } = window.location
+
+            const mergedSearchParams = mergeSearch(editorSearch, userSearch)
+            const uniqueSearchKeys = new Set(mergedSearchParams.keys())
+
+            let queryString = ''
+
+            const queryParams = Array.from(uniqueSearchKeys).flatMap((key) => {
+              const values = mergedSearchParams.getAll(key)
+              return values.map((value) => `${key}=${value}`)
+            })
+
+            if (queryParams.length) {
+              queryString = `?${queryParams.join('&')}`
+            }
 
-            const { search, hash } = window.location
             const copiedRenderProps = {
               ...renderProps,
-              to: `${hrefProp}${search}${hash}`,
+              to: `${previewPathName}${queryString}${userHash || editorHash}`,
             }
             delete (copiedRenderProps as Record<string, unknown>).href
 
-            return h(RouterLink, copiedRenderProps, children)
+            return h(RouterLink, copiedRenderProps, () => children)
```

Comment:
I noticed that there's a warning says the default slot content of RouterLink should be a function (both in preview and live mode), so I change this (and the one in `live-comonent.ts`) from `children` to `() => children`, which seems to fix the issue.

![Screenshot 2023-11-07 at 1 57 16 PM](https://github.com/samatechtw/pubstudio-builder/assets/59676941/6219096d-30b3-4bfa-837c-581c8ff21100)

Reply:
Oh yeah I remember seeing that! I think it's a strange warning though, I wonder if it's really more efficient.